allium/lib/site_generator.py

```
@@ -207,17 +319,43 @@ def generate_site(relay_set, args, progress_logger):
         )
 
+    # Protect crawlers from silently regressing to multi-megabyte HTML pages.
+    oversized = oversized_html_files(args.output_dir)
+    if oversized:
+        preview = ", ".join(
+            f"{path} ({size:,} bytes)" for path, size in oversized[:10]
+        )
+        raise RuntimeError(
+            f"{len(oversized)} generated HTML file(s) exceed the "
+            f"1,900,000-byte crawl-size guard: {preview}"
+        )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)